woocommerce/woocommerce-ios
# Home Screen Update: added the new tab "More" under a feature flag

### Description
As part of the home screen update project #5509 milestone 1, we are going to implement a new tab menu, which will substitute the Reviews tab and that will be used as a hub menu where we are going to implement reviews, coupons, etc...
In this PR, I implemented the feature flag for the hub menu, and we are going to display the new tab with an empty view just if the feature flag is enabled.

## Changes
- Added a feature flag `hubMenu`
- New image `hubMenu`
- New `HubMenuCoordinator` which will substitute the `ReviewsCoordinator` soon.
- New `HubMenuViewController` which conform to `UIHostingController` for presenting an istance of the SwiftUI view `HubMenu`
- Modified the `MainTabBarController`, introducing the new tab element and also the logic behind the feature flag. Updated tests.

### Testing instructions
1) Launch the app in debug mode or with the feature flag enabled.
2) You should be able to see the new tab.
3) Launch the app with the feature flag disabled.
4) The "Menu" tab should be hidden (you should see just 4 tabs).

### Screenshots
| Feature flag disabled             |  Feature flag enabled |
:-------------------------:|:-------------------------:
![Simulator Screen Shot - iPhone 13 Pro - 2021-11-29 at 15 08 25](https://user-images.githubusercontent.com/495617/143882550-17d8c0f4-e298-40dd-ac2b-c01e93583d57.png) | ![Simulator Screen Shot - iPhone 13 Pro - 2021-11-29 at 15 03 18](https://user-images.githubusercontent.com/495617/143882596-3b77f6e9-00af-4c11-8998-5dc8bad757c9.png)


---
